optaplanner-quickstarts/quarkus-school-timetabling/src/test/java/org/acme/schooltimetabling/solver/TimeTableConstraintProviderTest.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.acme.schooltimetabling.solver;
+
+import java.time.DayOfWeek;
+import java.time.LocalTime;
+
+import org.acme.schooltimetabling.domain.Lesson;
+import org.acme.schooltimetabling.domain.Room;
+import org.acme.schooltimetabling.domain.TimeTable;
+import org.acme.schooltimetabling.domain.Timeslot;
+import org.junit.jupiter.api.Test;
+import org.optaplanner.test.api.score.stream.ConstraintVerifier;
+
+public class TimeTableConstraintProviderTest {
+
+    // TODO Inject the constraintVerifier
```

Comment:
Yes, but not with `@SpringBootTest`. Than requires setting up the Spring Context, turning off web environment, ... and still is a big overhead. Unit tests need to be fast (milliseconds).

My proposal: inject it with a JUnit 5 extension. That works independently of Quarkus or Spring Boot. I have a POC.

Reply:
Or we don't make simple things complicated and leave it the way it is.

Reply:
Yes, that's also an option.

See #815 for the POC.

Reply:
The ConstraintVerifier is build from a subset of the SolverConfig information.
The Quarkus and Spring Boot code both build the SolverConfig in their own way (although in theory they have the same resulting behavior, as far as possible). Quarkus does this at build time, Spring at bootstrap runtime.
I want to make sure the unit test tests what is used in production, therefore it must use the SolverConfig build by Q/SB.

Reply:
For example: they might have an application.properties that switches from DROOLS to BAVET, or a solverConfig.xml that provides a custom property to the ConstraintProvider class.

And sooner or later people will want the ConstraintProvider class to be injected from the context I bet/fear...